### PR TITLE
fix(richtext): RTE: Local Context Menu with a selected style option stays visible after opening an Admin Menu form (#577)

### LINKED ISSUE #577: RTE: Local Context Menu with a selected style option stays visible after opening an Admin Menu form
**Steps to reproduce:**

1. open Edit env, /flow-container/ 
2. switch to Edit mode
3. add Full RTE
4. type any text in the RTE
5. highlight any text to get Local Context Menu
6. select any option, e.g. 'Superscript'
7. click File (or any icon in Admin Menu that has a form)

**Expected Results:**
Context Menu closes

**Actual Results:**
RTE context menu is still visible
RTEMenu

**NOTE:**
Context Menu closes as expected if no style in it selected

### Affected Version
reproduced in Master

### PR DESCRIPTION
**Pull request by marcopagliarulo**
## Changes
onBlur event added to the rich text editor's Hover menu. The hover menu is hidden when the focus is moved out it.

## Test Instructions
1. open Edit env, /flow-container
2. switch to Edit mode
3. add Full RTE
4. type any text in the RTE
5. highlight any text to get Local Context Menu
6. select any option, e.g. 'Superscript'
7. click File (or any icon in Admin Menu that has a form)

**Expected Results:**
Context Menu closes

## Related Issues
Fixes #577

### CODE REVIEW

**Review by VancheeZze — DISMISSED** [review]

**Comment by TatsianaMiraniuk** [comment]
Hi @marcopagliarulo ,

the issue is reproducible if a user select 'Link' style:
Link-ContextMenu

That can be seen in Full, Basic RTEs.
And also if Superscript is selected in Main Menu, CTA (linked elements)

MainMenuContext


Could you please take a look?


Thanks,
Tatsiana

**Review by VancheeZze — DISMISSED** [review]

File: packages/bodiless-richtext/src/components/TextSelectorButton.tsx [hunk 1]
```diff
@@ -33,53 +33,52 @@ const NodeSelectorButton = (props: ButtonProps) => {
...
 
-export const TextSelectorContext = React.createContext({ onClose: () => {} });
-
 const TextSelectorButton = ({
   children,
 }:props) => {
   const [visible, setVisible] = useState(false);
   const { Overlay, TextSelectorWrapper } = getUI(useUI());
-  const textSelectorContextValue = { onClose: () => setVisible(false) };
+  console.log(children);
```
Comment (VancheeZze) [inline packages/bodiless-richtext/src/components/TextSelectorButton.tsx:45]: Let's remove this console.log

File: packages/bodiless-richtext/src/components/TextSelectorButton.tsx [hunk 2]
```diff
@@ -33,53 +33,52 @@ const NodeSelectorButton = (props: ButtonProps) => {
...
+  const nodeSelectorProps = {
+    className: visible ? 'active bl-active node-selector-button' : '',
```
Comment (VancheeZze) [inline packages/bodiless-richtext/src/components/TextSelectorButton.tsx:47]: I don't think we need `active` class here. Let's only keep `bl-active`.

File: packages/bodiless-richtext/src/components/TextSelectorButton.tsx (hunk 1 above)
Comment (marcopagliarulo) [reply packages/bodiless-richtext/src/components/TextSelectorButton.tsx]: removed, thanks

File: packages/bodiless-richtext/src/components/TextSelectorButton.tsx (hunk 2 above)
Comment (marcopagliarulo) [reply packages/bodiless-richtext/src/components/TextSelectorButton.tsx]: fixed, thanks

**Review by VancheeZze — APPROVED** [review]

**Comment by TatsianaMiraniuk** [comment]
@marcopagliarulo ,

**issue_3** is still reproducible, can you please recheck?

**Steps to reproduce:**
1. open Edit env, any page
2. switch to Edit mode
3. in Main Menu, add an item
4. type any text in the item
5. highlight any text to get Local Context Menu
6. select 'superscript' option
7. click File (or any icon in Admin Menu that has a form)

**Expected Results:**
Context Menu closes

**Actual Results:**
Context Menu stays visible
Screen Shot 2020-12-03 at 3 50 24 PM

**Review by dewen — APPROVED** [review]

**Comment by beliayeu** [comment]
closing in favor of https://github.com/johnsonandjohnson/Bodiless-JS/pull/795
cc: @TatsianaMiraniuk , @marcopagliarulo , @dewen

**Pull request closed without merging**